lucene/core/src/java/org/apache/lucene/search/FuzzyQuery.java

```diff
@@ -99,7 +103,24 @@ public FuzzyQuery(Term term, int maxEdits, int prefixLength, int maxExpansions,
     this.prefixLength = prefixLength;
     this.transpositions = transpositions;
     this.maxExpansions = maxExpansions;
-    setRewriteMethod(new MultiTermQuery.TopTermsBlendedFreqScoringRewrite(maxExpansions));
+    if (term.text().length() == prefixLength) {
```

Comment:
I think the problem is that we're checking this in `getTermsEnum` and potentially returning a `SingleTermsEnum`. Checking it here and modifying rewriteMethod feels inconsistent.